File: CS/CustomRepositoryItem/fmDesigner.cs
using System;
using System.ComponentModel.Design;
using System.Windows.Forms;
using DevExpress.DataAccess.UI.Wizard.Services;
using DevExpress.XtraReports.UI;
// ...

namespace CustomRepositoryItem {
    public partial class fmDesigner : Form {
        readonly MyRepositoryItemsProvider myRepositoryItemsProvider;

        public fmDesigner() {
            InitializeComponent();
            myRepositoryItemsProvider = new MyRepositoryItemsProvider();
            ReplaceService(reportDesignerMDIController, typeof(IRepositoryItemsProvider), myRepositoryItemsProvider);
            reportDesignerMDIController.DesignPanelLoaded += reportDesignerMDIController_DesignPanelLoaded;
        }

        void reportDesignerMDIController_DesignPanelLoaded(object sender, DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs e) {
            ReplaceService(e.DesignerHost, typeof(IRepositoryItemsProvider), myRepositoryItemsProvider);
        }

        void ReplaceService(IServiceContainer container, Type serviceType, object serviceInstance) {
            if(container.GetService(serviceType) != null)
                container.RemoveService(serviceType);
            container.AddService(serviceType, serviceInstance);
        }

        void fmDesigner_Load(object sender, EventArgs e) {
            XtraReport report = new XtraReport1();
            reportDesignerMDIController.OpenReport(report);
        }
    }
}

File: VB/CustomRepositoryItem/fmDesigner.vb
Imports System
Imports System.ComponentModel.Design
Imports System.Windows.Forms
Imports DevExpress.DataAccess.UI.Wizard.Services
Imports DevExpress.XtraReports.UI
' ...

Namespace CustomRepositoryItem
    Partial Public Class fmDesigner
        Inherits Form

        Private ReadOnly myRepositoryItemsProvider As MyRepositoryItemsProvider

        Public Sub New()
            InitializeComponent()
            myRepositoryItemsProvider = New MyRepositoryItemsProvider()
            ReplaceService(reportDesignerMDIController, GetType(IRepositoryItemsProvider), myRepositoryItemsProvider)
            AddHandler reportDesignerMDIController.DesignPanelLoaded, AddressOf reportDesignerMDIController_DesignPanelLoaded
        End Sub

        Private Sub reportDesignerMDIController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DevExpress.XtraReports.UserDesigner.DesignerLoadedEventArgs)
            ReplaceService(e.DesignerHost, GetType(IRepositoryItemsProvider), myRepositoryItemsProvider)
        End Sub

        Private Sub ReplaceService(ByVal container As IServiceContainer, ByVal serviceType As Type, ByVal serviceInstance As Object)
            If container.GetService(serviceType) IsNot Nothing Then
                container.RemoveService(serviceType)
            End If
            container.AddService(serviceType, serviceInstance)
        End Sub

        Private Sub fmDesigner_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim report As XtraReport = New XtraReport1()
            reportDesignerMDIController.OpenReport(report)
        End Sub
    End Class
End Namespace
